Src/WitsmlExplorer.Api/Workers/AnalyzeGapWorker.cs

```diff
@@ -0,0 +1,181 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.Extensions.Logging;
+using Witsml.Data;
+using Witsml.Data.Curves;
+using Witsml.Extensions;
+using Witsml.ServiceReference;
+
+using WitsmlExplorer.Api.Jobs;
+using WitsmlExplorer.Api.Models;
+using WitsmlExplorer.Api.Models.Reports;
+using WitsmlExplorer.Api.Query;
+using WitsmlExplorer.Api.Services;
+using Index = Witsml.Data.Curves.Index;
+
+namespace WitsmlExplorer.Api.Workers;
+
+/// <summary>
+/// Worker for analyzing gaps for curves of log data.
+/// </summary>
+public class AnalyzeGapWorker : BaseWorker<AnalyzeGapJob>, IWorker
+{
+    private const string DataSeparator = ",";
+    public JobType JobType => JobType.AnalyzeGaps;
+    public AnalyzeGapWorker(ILogger<AnalyzeGapJob> logger, IWitsmlClientProvider witsmlClientProvider) : base(witsmlClientProvider, logger) { }
+    
+    /// <summary>
+    /// Find all gaps for selected mnemonics and required size of gap. If isn't select mnemonics, then are used all mnemonics on log for finding gaps. 
```

Comment:
Rewrite "If isn't select mnemonics, then are used all mnemonics on log for finding gaps." to "If no mnemonics are selected, all mnemonics on the log will be analyzed."

Reply:
fixed